azurerm/resource_arm_container_registry.go

```diff
@@ -227,8 +311,53 @@ func resourceArmContainerRegistryUpdate(d *schema.ResourceData, meta interface{}
 	return resourceArmContainerRegistryRead(d, meta)
 }
 
+func getReplicationLocationsToDelete(d *schema.ResourceData) []string {
+	oldGeoreplifationLocations, newGeoreplicationLocations := d.GetChange("georeplication_locations")
```

Comment:
Also, consider during this time if the SKU changes from premium to basic, do you need to remove the replications or does the change in sku automatically drop the replications?

Reply:
When updating an SKU from Premium to Basic, it automatically deletes all the replications. This is covered by tests.